main.tf

```diff
@@ -0,0 +1,166 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.16"
+    }
+  }
+
+  required_version = ">= 1.5.7"
+
+  backend "s3" {
+    bucket               = "ccao-terraform-state-us-east-1"
+    key                  = "workspaces/default/terraform.tfstate"
+    region               = "us-east-1"
+    workspace_key_prefix = "workspaces"
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
+locals {
+  s3_prefix                = "scripts/sales-val"
+  s3_bucket_data_warehouse = terraform.workspace == "prod" ? "ccao-data-warehouse-us-east-1" : aws_s3_bucket.data_warehouse[0].id
+  s3_bucket_glue_assets    = terraform.workspace == "prod" ? "ccao-glue-assets-us-east-1" : aws_s3_bucket.glue_assets[0].id
+  glue_job_name            = "sales_val_flagging${terraform.workspace == "prod" ? "" : "_${terraform.workspace}"}"
+  glue_crawler_name        = "ccao-data-warehouse-sale-crawler${terraform.workspace == "prod" ? "" : "-${terraform.workspace}"}"
+}
+
+variable "iam_role_arn" {
```

Comment:
[Variables](https://developer.hashicorp.com/terraform/language/values/variables) are values that must either be passed in by the Terraform caller, or set with a `default`. We keep the `iam_role_arn` out of version control because it contains a reference to our AWS Account ID, which [is not technically a secret but should be used cautiously](https://docs.aws.amazon.com/accounts/latest/reference/manage-acct-identifiers.html). The IAM role itself is not managed by this Terraform config because it is shared by other Glue jobs, so it will be defined in our upcoming org-wide Terraform config.